src/TuDa.CIMS.Api/Controllers/WorkingGroupController.cs

```diff
@@ -0,0 +1,93 @@
+    [HttpGet]
+    public async Task<IActionResult> GetAllAsync()
+    {
+        return (await _workingGroupService.GetAllAsync()).Match<IActionResult>(
+            value => Ok(value),
+            err => BadRequest(err)
+        );
+    }
+
+    /// <summary>
+    /// Retrieves a Working Group with the specific id from the service and returns it in an appropriate HTTP response.
+    /// </summary>
+    /// <param name="id">the unique id of the Working Group</param>
+    /// <returns> a 200 OK response if the operation is successfully and a 400 BadRequest response if any error occurs </returns>
+    [HttpGet($"{{{nameof(id)}:guid}}")]
+    public async Task<IActionResult> GetOneAsync(Guid id)
+    {
+        return (await _workingGroupService.GetOneAsync(id)).Match<IActionResult>(
+            value => Ok(JsonSerializer.Serialize(value)),
+            err => BadRequest(err)
+        );
+    }
+
+    /// <summary>
+    /// Updates an existing Working Group by its ID using the provided update model.
+    /// If the update is successful, returns a 200 OK response. If an error occurs during the update, an appropriate error response is returned.
+    /// </summary>
+    /// <param name="id">the unique id of the Working Group</param>
+    /// <param name="updateModel">the model containing the updated values for the Working Group </param>
+    [HttpPatch($"{{{nameof(id)}:guid}}")]
+    public async Task<IActionResult> UpdateAsync(Guid id, UpdateWorkingGroupDto updateModel)
+    {
+        return (await _workingGroupService.UpdateAsync(id, updateModel)).Match<IActionResult>(
+            _ => Ok(),
+            err => BadRequest(err)
+        );
+    }
+
+    /// <summary>
+    /// Deletes an existing Working Group by its ID.
+    /// If the removal is successful, returns a 200 OK response.
+    /// If an error occurs during the deletion, an appropriate error response is returned.
+    ///</summary>
+    /// <param name="id">the unique id of the Working Group</param>
+    [HttpDelete($"{{{nameof(id)}:guid}}")]
+    public async Task<IActionResult> RemoveAsync(Guid id)
+    {
+        return (await _workingGroupService.RemoveAsync(id)).Match<IActionResult>(
+            _ => Ok(),
+            err => BadRequest(err)
+        );
+    }
+
+    /// <summary>
+    /// Creates a new Working Group using the provided create model.
+    /// If the removal is successful, returns a 201 Created response.
+    /// If an error occurs during the deletion, an appropriate error response is returned.
+    /// </summary>
+    /// <param name="createModel">the model containing the data for the new Working Group</param>
+    [HttpPost]
+    public async Task<IActionResult> CreateAsync(CreateWorkingGroupDto createModel)
+    {
+        var result = await _workingGroupService.CreateAsync(createModel);
+
+        return result.Match<IActionResult>(
+            _ => CreatedAtAction(String.Empty, null),
```

Comment:
🔧 🌱 We should return something here. Maybe the GUID of the created group or the hole object

Reply:
```suggestion
            value => CreatedAtAction(String.Empty, value),
```

is that good enough?

Reply:
No because you would need to return the working group in repo and service. When this is done this will work

Reply:
How relevant was that again? 

I've made following changes:

`Task<ErrorOr<WorkingGroup>> CreateAsync(CreateWorkingGroupDto createModel);` in Repo
`Task<ErrorOr<WorkingGroup>> CreateAsync(CreateWorkingGroupDto createModel);` in Service

Reply:
This should be working. But it was not that important